Mathlib/Algebra/Order/Module/Algebra.lean

```diff
@@ -70,3 +70,44 @@ variable [SMulPosReflectLT α β]
 
 end SMulPosStrictMono
 end StrictOrderedSemiring
+
+namespace Mathlib.Meta.Positivity
+open Lean Meta Qq Function
+
+/-- Extension for `algebraMap`. -/
+@[positivity algebraMap _ _ _]
+def evalAlgebraMap : PositivityExt where eval {u β} _zβ _pβ e := do
+  let ~q(@algebraMap $α _ $instα $instβ $instαβ $a) := e | throwError "not `algebraMap`"
+  let pα ← synthInstanceQ (q(PartialOrder $α) : Q(Type u_1))
+  match ← core q(inferInstance) pα a with
+  | .positive pa =>
+    try
+      let _instαring ← synthInstanceQ q(OrderedCommSemiring $α)
+      let _instβring ← synthInstanceQ q(StrictOrderedSemiring $β)
+      let _instαβsmul ← synthInstanceQ q(SMulPosStrictMono $α $β)
+      assertInstancesCommute
+      return .positive q(algebraMap_pos $β $pa)
+    catch _ =>
+      let _instαring ← synthInstanceQ q(OrderedCommSemiring $α)
```

Comment:
Can you not synthesize this earlier so that you don't have to resynth it on the `catch` branch? maybe earlier because you need it for all branches?

Reply:
or is the issue here that you don't want to synth it for a `nonzero` branch?

Reply:
Yes exactly. I don't need to synthesize this in the `nonzero` branch